File: src/Blazor.Extensions.Toastr.JS/src/Toast.ts
export interface ToastElement extends HTMLDivElement {
  hide: () => void;
}

export type Toast = {
  title: string;
  text: string;
  type: ToastType;
  icon: string;
  timeout: number;
  onClick: () => void;
}

export enum ToastType {
  Info = 1,
  Success = 2,
  Warning = 3,
  Error = 4
}

File: src/Blazor.Extensions.Toastr/Models/Toast.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Blazor.Extensions
{
    public class Toast
    {
        #region Properties
        public string Title { get; set; }
        public string Text { get; set; }
        public ToastType? Type { get; set; } = ToastType.Info;
        public string Icon { get; set; }
        public int? Timeout { get; set; } = 4500;
        public Action Callback { get; set; }
        #endregion
    }
}